src/octopeer-github/content/DatabaseAdaptable/MessageSendDatabaseAdapter.ts

```diff
@@ -5,17 +5,40 @@
 /**
  * This class implements DatabaseAdaptable in a way that
  * the data is sent to the Chrome Extension Background page.
+ * @see ../../main/DatabaseAdaptable/ for more Documentation (copy-pasting is bad, mmkay)
  */
 class MessageSendDatabaseAdapter implements DatabaseAdaptable {
 
-    /**
-     * @see ../../main/DatabaseAdaptable
-     */
-    public post(data: IEventObject, success: Callback, failure: Callback) {
+    public postSemantic(data: ISemanticEvent, success: Callback, failure: Callback) {
         let postData: any = data;
         postData.elementID = (<ElementID>data.elementID).getElementID();
         postData.eventID = (<EventID>data.eventID).getEventID();
-        chrome.runtime.sendMessage(JSON.stringify(postData));
+        chrome.runtime.sendMessage(JSON.stringify({data: postData, type: "postSemantic"}));
+        success();
+    }
+
+    public postKeystroke(data: IKeystrokeEvent, success: Callback, failure: Callback) {
```

Comment:
Will do this in the next branch then :) In the worst case I'll just merge #67 in my branch before testing XD